### PR TITLE
Exporter zeroes timestamps when adding extension layers to the app image

### PR DESCRIPTION
**Pull request by natalieparellano**
### Summary

Before, we simply added the layer as output by kaniko (though we uncompressed the layer for daemon images).

After, we uncompress the layer so that we can read the tarball, then zero the mod time for each header. We can add the layer to the image uncompressed, as the specific imgutil layout/remote implementations will take care of compressing the layer where needed. The imgutil local package wants the layer uncompressed. 

We should investigate if it's possible to force kaniko not to do any compression - looking [here](https://github.com/GoogleContainerTools/kaniko/blob/11e2e9f40847754141ee8f1c009248facc651467/pkg/executor/build.go#L533-L535) and [here](https://github.com/google/go-containerregistry/blob/8dadbe76ff8c20d0e509406f04b7eade43baa6c1/pkg/v1/tarball/layer.go#L109) this seems intentionally not permitted.

#### Release notes

Exporter zeroes timestamps when adding extension layers to the app image

---

### Related

Resolves https://github.com/buildpacks/lifecycle/issues/1180

### CODE REVIEW

File: phase/exporter.go [hunk 1]
```diff
@@ -362,34 +357,6 @@ func isLocalImage(workingImage imgutil.Image) bool {
 	return false
 }
 
-func uncompressLayerAt(layerPath string, toArtifactsDir string) (string, error) {
```
Comment (natalieparellano) [inline phase/exporter.go:365]: IDK why I added this function, we could just call `layer.Uncompressed()`

**Comment by natalieparellano** [comment]
IDK, I think Codecov is being too picky here (I even added a bunch of tests!)

File: phase/exporter.go [hunk 2]
```diff
@@ -281,54 +279,51 @@ func (e *Exporter) addExtensionLayers(opts ExportOptions) error {
...
-		if localImage {
-			var calculatedDiffID string
-			if calculatedDiffID, err = uncompressLayerAt(layerPath, artifactsDir); err != nil {
-				return err
-			}
-			if calculatedDiffID != layerHex.String() {
-				return fmt.Errorf("digest of uncompressed layer from %s does not match expected value; found %q, expected %q", layerPath, calculatedDiffID, layerHex.String())
-			}
-			layerPath = filepath.Join(artifactsDir, calculatedDiffID)
+		if !blobExists {
```
Comment (jabrown85) [inline phase/exporter.go:287]: If the blob doesn't exist we are skipping it and not throwing an error - is that expected? If so, maybe a comment explaining why would be good for future us!

**Review by jabrown85 — APPROVED** [review]
I added a question but looks good

File: phase/exporter.go (hunk 2 above)
Comment (natalieparellano) [inline phase/exporter.go:287]:
```suggestion
		if !blobExists {
		// the referenced blob is a run image (base image) layer, not an extension-provided layer
```

Comment (natalieparellano) [reply phase/exporter.go:287]: Added a comment in https://github.com/buildpacks/lifecycle/pull/1289/commits/51050acd241dc0150df004bf17f08218b0ecb2a3

**Pull request merged**